core/src/types.ts

```diff
@@ -84,14 +84,37 @@ export declare interface UniswapV2LpTokenCalleeConfig {
     token1: string;
 }
 
+interface ValueSlotAddressAndOffset {
+    slot: string;
+    offset: number;
+}
+
 export declare interface CollateralConfig {
     title: string;
     ilk: string;
     symbol: string;
     decimals: number;
     exchange: RegularCalleeConfig | UniswapV2LpTokenCalleeConfig;
+    oracle: CollateralPriceSourceConfig;
+}
+
+interface OracleConfigBase {
+    hasDelay: boolean;
+    slotPriceValueBeginsAtPosition: number;
+}
+
+export declare interface OracleCurrentAndNextPrices extends OracleConfigBase {
+    currentPriceSlotAddress: string;
+    nextPriceSlotAddress: string;
+}
+
+export declare interface OracleCurrentPriceOnly extends OracleConfigBase {
+    currentPriceSlotAddress: string;
+    currentPriceValiditySlotAndOffset: ValueSlotAddressAndOffset;
 }
```

Comment:
Do you want to explicitly specify types to
- depend on those types in the code instead of `'key' in oracle` checks
- refer to it in the readme for clarity?

```suggestion
export declare interface OracleCurrentAndNextPrices extends OracleConfigBase {
    type: 'OracleTypeOne'
    currentPriceSlotAddress: string;
    nextPriceSlotAddress: string;
}

export declare interface OracleCurrentPriceOnly extends OracleConfigBase {
    type: 'OracleTypeTwo'
    currentPriceSlotAddress: string;
    currentPriceValiditySlotAndOffset: ValueSlotAddressAndOffset;
}
```

(just a suggestion)